tsdb/wlog/watcher.go

```diff
@@ -421,9 +453,30 @@ func (w *Watcher) watch(segmentNum int, tail bool) error {
 
 			return nil
 
+		// we haven't read due to a notification in quite some time, try reading anyways
 		case <-readTicker.C:
+			level.Debug(w.logger).Log("msg", "Watcher is reading the WAL due to timeout, haven't received any write notifications recently", "timeout", readTimeout)
 			err = w.readSegment(reader, segmentNum, tail)
+			readTicker.Reset(readTimeout)
```

Comment:
Is it necessary to reset the ticker here? It would have just triggered so should be a full 15 seconds until the next trigger.

Reply:
IMO we still should reset, the read could take some amount of time. I don't think calling `reset` is expensive in some way either?

I'm not opposed to removing it either.

Reply:
🤷 leaving it in seems fine, it's an edge case anyway.